src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1090,9 +1091,13 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             bool skipUsesIsNullable,
             MessageID? experimentalFeature)
         {
+            var syntaxTrees = source.GetSyntaxTrees(parseOptions, sourceFileName);
+
             if (options == null)
             {
-                options = TestOptions.ReleaseDll;
+                bool hasTopLevelStatements = syntaxTrees.Any(s => s.GetRoot().DescendantNodes().OfType<GlobalStatementSyntax>().Any());
```

Comment:
>DescendantNodes() [](start = 78, length = 17)

Does this look too deep? I think we should check only immediate children of a compilation unit.
See
```
        internal static bool IsTopLevelStatement([NotNullWhen(true)] GlobalStatementSyntax? syntax)
        {
            return syntax?.Parent?.IsKind(SyntaxKind.CompilationUnit) == true;
        }
```